### ISSUE TITLE
BGE QNN Failure to Run

### ISSUE THREAD

**Issue by brian-at-pieces** [microsoft/Olive#1721]
**Describe the bug**
I'm trying to use the BGE embedding model with onnxruntime + Snapdragon NPU. I can successfully run the optimization script from the [BGE example](https://github.com/microsoft/Olive/tree/52b903e080696db299aa8d37ef0c78d1caa8d46e/examples/bge), but when trying to run the output model I'm getting op errors for Unsqueeze, Expand, and Cast which are the first few layers the attention_mask input flows into.

<img width="938" alt="Image" src="https://github.com/user-attachments/assets/75e38b0f-a203-4e55-af93-967f94bce81a" />

**To Reproduce**
1. Install Python x64
2. Install Olive-qnn, onnxruntime-qnn
3. Run olive optimization script for BGE above
4. Run simple inference example with output model and Python arm64

**Expected behavior**
Goode model run

**Olive config**
{
    "input_model": {
        "type": "HfModel",
        "model_path": "BAAI/bge-small-en-v1.5",
        "task": "feature-extraction",
        "io_config": {
            "input_names": [ "input_ids", "attention_mask", "token_type_ids" ],
            "input_shapes": [ [ 1, 128 ], [ 1, 128 ], [ 1, 128 ] ],
            "input_types": [ "int64", "int64", "int64" ],
            "output_names": [ "last_hidden_state", "state" ]
        }
    },
    "systems": {
        "local_system": {
            "type": "LocalSystem",
            "accelerators": [ { "device": "npu", "execution_providers": [ "QNNExecutionProvider" ] } ]
        }
    },
    "data_configs": [
        {
            "name": "quantize_data_config",
            "type": "HuggingfaceContainer",
            "load_dataset_config": { "data_name": "mteb/banking77", "split": "test" },
            "pre_process_data_config": { "max_length": 128, "padding": "max_length", "input_cols": [ "text" ] },
            "dataloader_config": { "batch_size": 1 }
        }
    ],
    "evaluators": {
        "common_evaluator": {
            "metrics": [
                {
                    "name": "accuracy",
                    "type": "custom",
                    "sub_types": [
                        {
                            "name": "accuracy_custom",
                            "priority": 1,
                            "higher_is_better": true,
                            "goal": { "type": "max-degradation", "value": 0.05 }
                        }
                    ],
                    "user_config": {
                        "user_script": "user_script.py",
                        "evaluate_func": "eval_accuracy",
                        "evaluate_func_kwargs": { "tasks": [ "Banking77Classification" ] }
                    }
                },
                {
                    "name": "latency_qnn",
                    "type": "latency",
                    "data_config": "quantize_data_config",
                    "sub_types": [ { "name": "avg" }, { "name": "max" }, { "name": "min" } ]
                }
            ]
        }
    },
    "passes": {
        "conversion": { "type": "OnnxConversion", "target_opset": 17 },
        "dynamic_shape_to_fixed": {
            "type": "DynamicToFixedShape",
            "dim_param": [ "batch_size", "sequence_length" ],
            "dim_value": [ 1, 128 ]
        },
        "surgery": { "type": "GraphSurgeries", "surgeries": [ { "surgeon": "ReplaceAttentionMaskValue" } ] },
        "QNNPreprocess": { "type": "QNNPreprocess", "fuse_layernorm": true },
        "OnnxQuantization": {
            "type": "OnnxQuantization",
            "data_config": "quantize_data_config",
            "activation_type": "QUInt16",
            "weight_type": "QUInt8",
            "calibrate_method": "MinMax",
            "quant_preprocess": true
        }
    },
    "host": "local_system",
    "target": "local_system",
    "cache_dir": "cache",
    "output_dir": "models/bge-small-en-v1.5",
    "evaluate_input_model": false
}

**Olive logs**
Add logs here.

**Other information**
 - OS: Windows Arm64
 - Olive version: 0.9.0.dev0
 - ONNXRuntime package and version: onnxruntime-qnn 1.22.0.dev20250321002
 - Transformers package version: 4.50.3

**Additional context**

**Comment by xieofxie** [comment]
did you set disable cpu fallback? did the evaluator work in olive?

**Comment by brian-at-pieces** [comment]
Yes I disabled CPU fallback. And the evaluator was actually failing, but I assumed that was an issue with having to use the x64 onnxruntime-qnn package for optimization whereas the arm64 package [should be used for inference.](https://onnxruntime.ai/docs/execution-providers/QNN-ExecutionProvider.html#:~:text=Install%20the%20ONNX%20Runtime%20x64%20python%20package.%20(please%20note%2C%20you%20must%20use%20x64%20package%20for%20quantizing%20the%20model.%20use%20the%20arm64%20package%20for%20inferencing%20and%20utilizing%20the%20HTP/NPU))

**Comment by xieofxie** [comment]
it is possible that not all nodes are supported. When i created this olive json, I allow CPU fallback, so I think that's the issue. could you try with allowing cpu fallback again?

**Comment by brian-at-pieces** [comment]
Enabling CPU fallback actually slows down inference a bit. Averaging over 50 runs, the speed is ~27ms with NPU + CPU fallback and ~20ms with just CPU. Guessing this is due to passing data between NPU and CPU.

Assuming I've tested everything properly, I'd say that the [BGE optimization example](https://github.com/microsoft/Olive/tree/52b903e080696db299aa8d37ef0c78d1caa8d46e/examples/bge) is pretty misleading.

**Comment by brian-at-pieces** [comment]
Okay so interestingly I re-ran the optimizer script but changed the input size from 128 to 512 tokens and now I'm seeing massive speedups on the NPU! CPU-only is now ~187ms and NPU + CPU fallback is ~77ms. A 2.4x speedup isn't too shabby 😁 Unless there was some hiccup, my thought is that for smaller inputs backing off to the CPU has a more pronounced impact on overall latency, but as input size grows the benefits of the NPU outweigh this

**Comment by xieofxie** [comment]
Good to know it! Yes, when I created the sample, I didn't measure the latency and I only make sure the qdq model is small and accuracy is ok. so sorry for the misleading
